packages/client/src/links/httpBatchStreamLink.ts

```diff
@@ -7,13 +7,16 @@ import { streamingJsonHttpRequester } from './internals/parseJSONStream';
 import { Operation } from './types';
 
 const streamRequester: RequesterFn = (requesterOpts) => {
+  const textDecoder =
```

Comment:
I'd use a helper like we do for `fetch` https://github.com/trpc/trpc/blob/df2d2fb3cbbe5736ee6e6594d7b2bd01bd6c112d/packages/client/src/getFetch.ts#L12-L28 so it's more clear what went wrong in case this fails.